archunit/src/main/java/com/tngtech/archunit/core/importer/ClassFileImportRecord.java

```diff
@@ -90,10 +93,18 @@ void setStaticInitializer(String ownerName, DomainBuilders.JavaStaticInitializer
         staticInitializerBuildersByOwner.put(ownerName, builder);
     }
 
-    void addAnnotations(String ownerName, Set<DomainBuilders.JavaAnnotationBuilder> annotations) {
+    void addClassAnnotations(String ownerName, Set<DomainBuilders.JavaAnnotationBuilder> annotations) {
         this.annotationsByOwner.putAll(ownerName, annotations);
     }
 
+    public void addMemberAnnotations(String declaringClassName, String memberName, String descriptor, Set<DomainBuilders.JavaAnnotationBuilder> annotations) {
+        this.annotationsByOwner.putAll(getMemberKey(declaringClassName, memberName, descriptor), annotations);
+    }
+
+    void addAnnotationDefaultValue(String declaringClassName, String methodName, String descriptor, DomainBuilders.JavaAnnotationBuilder.ValueBuilder valueBuilder) {
+        annotationDefaultValuesByOwner.put(getMemberKey(declaringClassName, methodName, descriptor), valueBuilder);
```

Comment:
The owner you definitely need, same reason. The `descriptor` ... well, to be honest that's not necessary :wink: Cause the JLS does not allow annotation method parameters and there can't be two methods with the same name and signature but different return types within the same type.
If you don't have a strong objection, I would still keep it this way though, because

* I did some profiling and I can not see a performance difference whether adding the descriptor to the key or not
* this is still a "method", even though we know it cannot have parameters

But I would just go for consistency if there is no reason to deviate, i.e. simply have one "member key" format. And in all other contexts the descriptor is important. I don't know if it really makes the code easier to understand if we add the context knowledge here, that there can't be two methods with the same name, because they are declared within annotations and the JLS forbids it. It also looks the same in the ASM context... A method with a descriptor. So coming from the other side it could also look surprising to ignore the descriptor, while we need it in all other cases.